### DISCUSSION TITLE
"This PR was configured for branch automerge, however this is not possible so it has been raised as a PR instead."

### DISCUSSION THREAD

**Discussion by janpio** [renovatebot/renovate#8762] [Request Help]
We are using `  "automergeType": "branch"` but still get a lot of PRs, where I can not figure out why the branch itself was not enough to merge to the main branch. Example: https://github.com/prisma/e2e-tests/pull/1452 Is there a way to figure out _why_ automerge did not work?

Logs are probably at https://app.renovatebot.com/dashboard#github/prisma/e2e-tests/304205069, but I could not find anything in there explaining _why_ the automerge did not work only a log of the comment being added.

**Comment by viceice** [comment]
can you paste the logs here, i think there is a relevant message in the logs, maybe debug only

**Reply by janpio** [reply]
[renovate.log](https://github.com/renovatebot/renovate/files/6004696/renovate.log)

Not suuuuper sure this is the correct run though as we are on the hosted GH app and I am not sure how to connect the comment to the right logfile.

**Reply by viceice** [reply]
```
DEBUG: Automerging branch(branch="renovate/webpack-5.x")
INFO: Failed to automerge branch(branch="renovate/webpack-5.x")
{
  "err": {
    "task": {
      "commands": [
        "merge",
        "--ff-only",
        "renovate/webpack-5.x"
      ],
      "format": "utf-8"
    },
    "message": "fatal: refusing to merge unrelated histories\n",
    "stack": "Error: fatal: refusing to merge unrelated histories\n\n    at GitExecutorChain.onFatalException (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:65:87)\n    at GitExecutorChain.<anonymous> (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:56:28)\n    at Generator.throw (<anonymous>)\n    at rejected (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:6:65)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)"
  }
}
DEBUG: mergeStatus=failed(branch="renovate/webpack-5.x")
DEBUG: Branch automerge not possible(branch="renovate/webpack-5.x")
{
  "mergeStatus": "failed"
}
```

Has someone done a force push on your `dev` branch?

**Reply by janpio** [reply]
Not that I am aware.

We have a GH Action that commits _some_ dependency updates, also in the same `package.json` file that was changed by Renovate. But as that is in a different line, should merge cleanly really: https://github.com/prisma/e2e-tests/commits/dev

**Reply by viceice** [reply]
ok, maybe you have a problem with multiple concurrent updates, as your error message clearly says there is something wrong with your git

**Reply by rarkins** [reply]
Do any branch automerges work?

**Reply by janpio** [reply]
Yes there are quite a few fortunately where we just have a direct commit left, not via a merged PR: https://github.com/prisma/e2e-tests/commit/910f73fef0ed8ba6c7572c291515af019507e627

**Reply by rarkins** [reply]
Did it try to automerge two in one run? Did it automerge any in the run prior to the run that failed?

**Reply by janpio** [reply]
> Did it try to automerge two in one run?

I can only see one `Automerging branch` in the logs of this run.



> Did it automerge any in the run prior to the run that failed?

No, the previous auto merge was a day ago. Around the time the "not possible" comment another dependency update was merged manually via a PR (without waiting for the PR to be automerged as I edited a part of the PR).

**Reply by rarkins** [reply]
If you committed to the main branch after Renovate started running but before it tried to automerge then that would explain the merge failure. Can you compare timestamps to confirm?

**Reply by rarkins** [reply]
But even if so, we should treat this as a "repository changed" error and not time up on the branch merge altogether

**Reply by janpio** [reply]
- Renovate log for Job ID `304205069` shows `Thursday, February 18th 2021, 6:07:34 pm`, not other timestamp unfortunately
- Comment was added: `2021-02-18T17:07:32Z` (so Renovate log time is probably when it was written [and not started])
- I manually merged a different Renovate PR a bit earlier: `Feb 18, 2021, 6:05 PM GMT+1`

Is there a way to see when the Renovate run started?

Update:
From the logs:
```
DEBUG: latest commit
{
  "latestCommitDate": "2021-02-18T18:05:48+01:00"
}
```
So it seems that commit had already happened when the Renovate run started.

**Reply by rarkins** [reply]
This will be fixed by #8796. The problem is that currently we assume that anyone configuring rebaseWhen=conflicted doesn't want us to rebase stale automerging branches.

**Reply by janpio** [reply]
Nice, that sounds like it should fix it for me. Will continue observing.

**Comment by rarkins** [comment]
I found a second case where a branch automerge was raised as a PR in this repo. Here's the git merge error:

```json
    "message": "Pushing to https://github.com/prisma/e2e-tests.git\nPOST git-receive-pack (369 bytes)\nremote: error: GH006: Protected branch update failed for refs/heads/dev.        \nremote: error: 4 of 54 required status checks have not succeeded: 1 expected and 3 failing.        \nerror: failed to push some refs to 'https://github.com/prisma/e2e-tests.git'\n",
```

Code inspection revealed we were (a) fetching only the default 30 per page, and (b) not paginating, so that explains that one. I will fix it today.

**Reply by SubJunk** [reply]
@rarkins I have another instance of this error, can you help? https://github.com/UniversalMediaServer/UniversalMediaServer/pull/3611#issuecomment-1314636627

**Reply by rarkins** [reply]
Check your logs
